target/scripts/build/packages.sh

```diff
@@ -0,0 +1,101 @@
+#! /bin/bash
+
+set -eE -u -o pipefail
+
+# shellcheck source=../helpers/log.sh
+source /usr/local/bin/helpers/log.sh
+
+export DEBIAN_FRONTEND=noninteractive
+QUIET='-qq' ; _log_level_is_trace && QUIET='-y'
+
+function _pre_installation_steps() {
+  _log 'info' 'Starting package installation'
+  _log 'debug' 'Running pre-installation steps'
+
+  _log 'trace' 'Updating package signatures'
+  apt-get "${QUIET}" update
+
+  _log 'trace' 'Installing packages that are needed early'
+  apt-get "${QUIET}" install --no-install-recommends \
+    apt-transport-https apt-utils ca-certificates \
+    curl gnupg2 ssl-cert 2>/dev/null
+
+  _log 'trace' 'Upgrading packages'
+  apt-get "${QUIET}" dist-upgrade
+
+  _log 'trace' 'Updating CA certificates'
+  update-ca-certificates
+}
+
+function _install_postfix() {
+  _log 'debug' 'Installing Postfix'
+
+  _log 'warn' "Applying workaround for Postfix bug (see https://github.com/docker-mailserver/docker-mailserver/issues/2023#issuecomment-855326403)"
+
+  mv /bin/hostname /bin/hostname.bak
+  echo "echo 'docker-mailserver.invalid'" >/bin/hostname
+  chmod +x /bin/hostname
+  apt-get "${QUIET}" install --no-install-recommends postfix
+  mv /bin/hostname.bak /bin/hostname
+}
+
+function _setup_apt_sources() {
+  _log 'debug' 'Acquiring package sources (PPAs)'
+
+  _log 'trace' 'Getting RSPAMD GPG keys and PPA'
+  local RSPAMD_GPG_KEY_LOCATION='/etc/apt/trusted.gpg.d/rspamd.gpg'
+  curl -sSfL https://rspamd.com/apt-stable/gpg.key \
+    | gpg --dearmor >"${RSPAMD_GPG_KEY_LOCATION}"
+  echo \
+    "deb [arch=amd64 signed-by=${RSPAMD_GPG_KEY_LOCATION}] http://rspamd.com/apt-stable/ bullseye main" \
+    >/etc/apt/sources.list.d/rspamd.list
+
+  _log 'trace' 'Updating package signatures again'
+  apt-get "${QUIET}" update
+}
```

Comment:
This is new and looks like it could be deferred to a separate PR (_better for `git blame` / context_).

My preference (_unless you know of an issue otherwise_) would be to have `rspamd` managed via it's own stage, like `clamav` and `fail2ban` should be. For now though, I think this is fine, but should be introduced via a separate PR.

Reply:
`rspamd` is not scope of that PR and should be done separately.

Reply:
Agreed!